src/plugins/discover/public/application/components/data_visualizer_grid/data_visualizer_grid.tsx

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useEffect, useMemo, useRef, useState } from 'react';
+import { Filter } from '@kbn/es-query';
+import { DataView, Query } from '../../../../../data/common';
+import { DiscoverServices } from '../../../build_services';
+import {
+  EmbeddableInput,
+  EmbeddableOutput,
+  ErrorEmbeddable,
+  IEmbeddable,
+  isErrorEmbeddable,
+} from '../../../../../embeddable/public';
+import { SavedSearch } from '../../../saved_searches';
+import { GetStateReturn } from '../../apps/main/services/discover_state';
+
+export interface DataVisualizerGridEmbeddableInput extends EmbeddableInput {
+  indexPattern: DataView;
+  savedSearch?: SavedSearch;
+  query?: Query;
+  visibleFieldNames?: string[];
+  filters?: Filter[];
+  showPreviewByDefault?: boolean;
+}
+export interface DataVisualizerGridEmbeddableOutput extends EmbeddableOutput {
+  showDistributions?: boolean;
+}
+
+export interface DiscoverDataVisualizerGridProps {
+  /**
+   * Determines which columns are displayed
+   */
+  columns: string[];
+  /**
+   * The used data view
+   */
+  indexPattern: DataView;
+  /**
+   * Discover plugin services
+   */
+  services: DiscoverServices;
+  /**
+   * Optional saved search
+   */
+  savedSearch?: SavedSearch;
+  /**
+   * Optional query to update the table content
+   */
+  query?: Query;
+  /**
+   * Filters query to update the table content
+   */
+  filters?: Filter[];
+  /**
+   * stateContainer to access and update app state preferences like to show preview or not
+   */
+  stateContainer?: GetStateReturn;
+}
+
+export const DiscoverDataVisualizerGrid = (props: DiscoverDataVisualizerGridProps) => {
+  const { services, indexPattern, savedSearch, query, columns, filters, stateContainer } = props;
+  const { uiSettings } = services;
+
+  const [embeddable, setEmbeddable] = useState<
+    | ErrorEmbeddable
+    | IEmbeddable<DataVisualizerGridEmbeddableInput, DataVisualizerGridEmbeddableOutput>
+    | undefined
+  >();
+  const embeddableRoot: React.RefObject<HTMLDivElement> = useRef<HTMLDivElement>(null);
+
+  const showPreviewByDefault = useMemo(
+    () =>
+      stateContainer ? !stateContainer.appStateContainer.getState().hideAggregatedPreview : true,
+    [stateContainer]
+  );
+
+  useEffect(() => {
+    embeddable?.getOutput$().subscribe((output: DataVisualizerGridEmbeddableOutput) => {
```

Comment:
I think that the `unsubscribe` operation is missing here

Reply:
Thanks for catching that! Fixed here https://github.com/elastic/kibana/pull/107184/commits/5f2a22803133b5cb7def597f237c41a0324f27d3